pkg/receive/hashring.go

```diff
@@ -103,27 +106,51 @@ func (p sections) Less(i, j int) bool { return p[i].hash < p[j].hash }
 func (p sections) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
 func (p sections) Sort()              { sort.Sort(p) }
 
+type ketamaEndpoint struct {
+	address string
+	az      string
+}
+
 // ketamaHashring represents a group of nodes handling write requests with consistent hashing.
 type ketamaHashring struct {
-	endpoints    []string
+	endpoints    []ketamaEndpoint
 	sections     sections
 	numEndpoints uint64
 }
 
+func getAZInfo(endpoint string) (string, string) {
+	s := strings.Split(endpoint, ",")
+	ep := s[0]
+	var az string
+	if len(s) == 1 {
+		az = ""
+	} else {
+		az = s[1]
+	}
+	return ep, az
```

Comment:
Maybe we can constrain to len 2 and avoid else?
```suggestion
	s := strings.SplitN(endpoint, ",", 2)
	ep := s[0]
	az := ""
	if len(s) == 2 {
		az = s[1]
	}
	return ep, az
```

Reply:
Ah yes, much cleaner. Thanks!